Test: Add DNS proxy test

eloycoto opened the pull request:
Two new tests: 

- Added a new test to make sure that the old TOCIDR are not deleted when a
new FQDN rule is added.
- Added a new test that uses DNS roundrobin to validate that the FQDN is working correctly. 

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6298)
<!-- Reviewable:end -->

eloycoto committed:
Test: Add DNS proxy TOCIDR test.

Added a new test to make sure that the old TOCIDR are not deleted when a
new FQDN rule is added.

Also added a new subdomain in bind (outside.test) to be able to test
not allowed domains.

Added monitor in test

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Test: Add roundrobin DNS test.

Add a new test that test dns names with multiples entries to validate
that the policy is applied correctly in all cases.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

tgraf commented on test/runtime/fqdn.go:
```diff
@@ -245,4 +304,67 @@ var _ = Describe("RuntimeFQDNPolicies", func() {
+			},
+			{
+				"toFQDNs": [{
+					"matchName": "cilium.test"
```
This is whitelisting `cilium.test` but we test to `world1.cilium.test`

eloycoto replied:
sorry, fix this in the next commit: 
https://github.com/cilium/cilium/pull/6298/commits/fcfd2fe5c97be1d0669f4b80da06326cb7109223

Will fix now.

tgraf reviewed (changes requested):
These tests look great! All tests seem to whitelist `cilium.test` but then test connectivity to `world.cilium.test` or similar. Shouldn't this break?

eloycoto committed:
FQDN: Validate that ToCIDRSet is entered when a ToFQNDS is in place.

Validate that the ToCIDRSet is in place when a TOFQDNS is in place to
avoid weird policy get issues.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>
